lib/iex/lib/iex/helpers.ex

```diff
@@ -802,8 +802,11 @@ defmodule IEx.Helpers do
           length
         end
 
-      IO.write(printer.(item, offset))
-      length + offset
+      printed = printer.(item, offset)
+      actual_offset = String.length(printed) + 1
```

Comment:
By recomputing the `offset` whenever the length is exceeding the expected target, we avoid this kind of issue:

```
iex(3)> ls
foobauhoiuhfsdiohfoihfajojfoiajfojasfojoicjoicjocjcoodfosdosejoe fseventsd-uuid                     powerlog
foo.bar
```

and get the following instead:

```
iex(2)> ls
foobauhoiuhfsdiohfoihfajojfoiajfojasfojoicjoicjocjcoodfosdosejoe fseventsd-uuid
powerlog                           foo.bar
```